### DISCUSSION TITLE
Slicing problems

### DISCUSSION THREAD

**Discussion by SaneSoup** [mlavik1/UnityVolumeRendering#260] [Q&A]
Hey there,

I am currently working on a little project and could need some help.

My journey so far:
I started using the vtk library in Python for a volume render but couldn't get the UI to work because vtk and tkinter didn't like each other.
Then I switched to Godot because of the python like syntax, I tried a voxel addon to create the volume but things didn't work out again.
Now I am using Unity to solve my problems.
I have dowloaded the VtkToUnityPlugin https://gitlab.com/3dheart_public/vtktounity and got the first halve of my program to work like intended.
The first halve involves a button, which lets you choose a folder with DICOM files and makes a volume with it, another button destroys the volume again and one can press the first button again.

The problem now:
My problem is the second halve of my intended program, I want to make moveable slices or at least one slice to view the inner parts of the loaded volume. I have already tried to use a shader, which makes the parts behind a certain plane invisible, however the volume does not posess a unity material because it is made with a plugin specific render-pipeline. There is a script in the provided assets called "VtkVolumeCropPlane" but it is stated nowhere how it works.

So I am thinking of starting from scracth using the code provided by @mlavik1 but I wanted to ask if my intentions are achieveable using his code or not before I start something new or if I can somehow integrate it into the VtkToUnityPlugin?

**Comment by mlavik1** [comment]
Hi! Looks like you went on a long journey though Python, Godot and Unity 😁 
I'm not at all familiar with VTK, so I'm unfortunately not able to give an answer on what can be done there. 
However, regarding UnityVolumeRendering: If what you want to do is create a slicing plane that can be moved through the volume, then that's already supported: See the [documentation for the slice renderer](https://github.com/mlavik1/UnityVolumeRendering/blob/master/Documentation/General/SliceRenderer/SliceRenderer.md). it creates a physical plane in the scene, that you can move around (you can optionally hide it from the scene).
If what you want is a cross section plane, that just cuts the volume in two, then see the [cross section tool documentation](https://github.com/mlavik1/UnityVolumeRendering/blob/master/Documentation/General/CrossSectionTools/CrossSectionTools.md).
It should also be possible to create a plane that works as both a slicing plane (for slice renderer) and cross section plane, by adding the two relevant components to it.

Regarding Godot, there's also a Godot version of this plugin: https://github.com/riccardolops/GodotVolumetricRendering
However, I don't think we implemented a slice renderer there yet. Will hopefully get back to that someday when I have time (™️ )

**Reply by SaneSoup** [reply]
Hey there, thanks for the reply.

I'll look into your code further in the next days.

+ Since you are propably more familiar with the DICOM file format than I am and If you don't mind to look at this question here: https://stackoverflow.com/questions/78882104/dicom-files-not-sorted Iwould be very happy.

**Reply by mlavik1** [reply]
I think maybe you sent me the wrong link? That's a news article about Disney 😁

**Reply by SaneSoup** [reply]
Whoopsie, strg + c never failed me like that before. Here should be the actual link https://stackoverflow.com/questions/78882104/dicom-files-not-sorted 😅
I should check my own links in the future before sending them 😂

**Reply by mlavik1** [reply]
Haha, no problem!
And no I'm not sure about what they mean by that error message.. Maybe the DICOM files are lacking some metadata?
Are you able to load them in Slicer? https://www.slicer.org/
If that works, you could export them from Slicer (as NRRD or DICOM) and see if that fixes it.

**Reply by SaneSoup** [reply]
Thanks for the response :)

I don't think that they are lacking metadata, they have more than the other files (that work). So they could possibly have too many?

I just tried loading them into the slicer and it worked, then I right-clicked the data and chose "export to file system".
![grafik](https://github.com/user-attachments/assets/d8c653da-e462-473c-9519-7488d992474f)
It did save them in a folder as DICOM files but when I tried to use them in unity and python the same error message appears sadly.

I am not familliar with this slicer application and used it for the first time so I might have done something wrong by exporting it like I mentioned above, if that's the case then please tell me 😅

Have a great day!
